cargo-culture-kit/src/rules/under_source_control.rs

```diff
@@ -0,0 +1,166 @@
+use super::{Rule, RuleContext, RuleOutcome};
+use std::path::Path;
+
+/// Rule that asserts a good Rust project:
+/// "Should be under source control"
+///
+/// # Justification
+///
+/// Source control, a.k.a. version control, is essential for the coordinated
+/// development of software projects.
+///
+/// # Caveats
+///
+/// The current implementation does a surface level check for the
+/// presence of hidden metadata subdirectories associated with popular
+/// version control systems:
+///
+/// * git
+/// * mercurial (hg)
+/// * svn
+/// * bazaar
+/// * darcs
+///
+#[derive(Debug, Default)]
+pub struct UnderSourceControl;
+
+const VC_SUBDIRS: &[&str] = &[".git", ".hg", ".bzr", ".svn", "_darcs"];
+
+impl Rule for UnderSourceControl {
+    fn description(&self) -> &str {
+        "Should be under source control."
+    }
+
+    fn evaluate(&self, context: RuleContext) -> RuleOutcome {
+        if AncestorDirs::from_file(context.cargo_manifest_file_path)
+            .any(|dir| VC_SUBDIRS.iter().any(|subdir| dir.join(subdir).is_dir()))
+        {
+            RuleOutcome::Success
+        } else {
+            RuleOutcome::Failure
+        }
+    }
+}
+
+struct AncestorDirs<'p> {
```

Comment:
Did you consider using https://doc.rust-lang.org/std/path/struct.Path.html#method.ancestors and requiring `nightly`?

Reply:
I strongly wanted to avoid a `nightly` requirement, or a requirement for a hyper-recent Rust in general.  The behavior here is a little bit different, also, in that it does not include the original file path in the iterator output.